addon/@ember/test-waiters/wait-for.ts

```diff
@@ -96,3 +135,33 @@ function isThenable(
     typeof maybePromise.then === 'function'
   );
 }
+
+function isGenerator(
+  maybeGenerator: any | CoroutineGenerator<unknown>
+): maybeGenerator is CoroutineGenerator<unknown> {
+  return typeof maybeGenerator[Symbol.iterator] === 'function';
```

Comment:
We can't rely on `Symbol` being defined (we still support IE11 without polyfills).

Reply:
How do we handle this then? I guess use duck typing to check for the generator interface, then? Check for `next`, `throw`, and `return` methods? Technically I suppose we could just check for a `next()` method, but I think I feel better about targeting this to generator functions (well, functions that return something that implements the generator object interface) rather than any function that returns something implementing the iterator interface.

Reply:
Related (from my big comment above) if we do proxy the generator object's `@@iterator` method in `waitForGenerator()`, how would we handle that in IE11? Do you know offhand, or should I fire up my Windows VM and look at the result of a transpiled generator function invocation?